src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -191,6 +191,12 @@ object Simplify {
           case c => Some(c)
         })
 
+      case MatrixUnionRows(children) if children.exists(_.isInstanceOf[MatrixUnionRows]) =>
+        MatrixUnionRows(children.flatMap {
+          case u: MatrixUnionRows => u.children
+          case c => Some(c)
+        })
+
```

Comment:
add two to invert MatrixRowsTable(MatrixUnionRows()) and same for MatrixColsTable?